core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -57,6 +57,31 @@ object ZIOSpec
           testM("returns success when it happens after failure") {
             assertM(ZIO.fail(42).raceAll(List(IO.succeed(24) <* live(ZIO.sleep(100.millis)))), equalTo(24))
           }
+        ),
+        suite("foreachPar")(
+          testM("runs effects in parallel") {
+            assertM(for {
+              p <- Promise.make[Nothing, Unit]
+              _ <- UIO.foreachPar(List(UIO.never, p.succeed(())))(a => a).fork
+              _ <- p.await
+            } yield true, equalTo(true))
+          },
+          testM("propagates error") {
+            val ints = List(1, 2, 3, 4, 5, 6)
+            val odds = ZIO.foreachPar(ints) { n =>
+              if (n % 2 != 0) ZIO.succeed(n) else ZIO.fail("not odd")
+            }
+            assertM(odds.flip, equalTo("not odd"))
+          },
+          testM("interrupts effects on first failure") {
+            val actions = List(
+              ZIO.never,
+              ZIO.succeed(1),
```

Comment:
Could you update some `Ref` with `delay` to check that this effect got interrupted (not updated the ref)?